common/compat.c

```diff
@@ -308,6 +308,8 @@ p11_mmap_open (const char *path,
 void
 p11_mmap_close (p11_mmap *map)
 {
+	if (map == NULL)
```

Comment:
I wouldn't introduce free-like semantics to `p11_mmap`; it should be an programming error if called on NULL.